### PR TITLE
Allow sorting assigned references

### PR DESCRIPTION
**Pull request by IKarbowiak**
Allow sorting assigned attribute values.
- Introduce intermediate models between `values` and `Assigned` models
- Split attribute models into small files
- Add mutations for sorting: `PageReorderAttributeValues`, `ProductVariantReorderAttributeValues`, `ProductReorderAttributeValues`

Linked task: [SALEOR-1853](https://app.clickup.com/t/2549495/SALEOR-1853)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1853 Allow sorting assigned references](https://app.clickup.com/t/2549495/SALEOR-1853)

File: saleor/attribute/migrations/0005_add_assignedattributevalues.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+from django.db import migrations, models
+
+
+def create_through_product_relations(apps, schema_editor):
+    AssignedProductAttribute = apps.get_model("attribute", "AssignedProductAttribute")
+    AssignedProductAttributeValue = apps.get_model(
+        "attribute", "AssignedProductAttributeValue"
+    )
+    instances = []
+    for assignment in AssignedProductAttribute.objects.all():
```
Comment (fowczarek) [inline saleor/attribute/migrations/0005_add_assignedattributevalues.py:13]: We should use iterator here. :policeman:

File: saleor/attribute/migrations/0005_add_assignedattributevalues.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    AssignedProductAttributeValue.objects.bulk_create(instances)
+
+
+def create_through_variant_relations(apps, schema_editor):
+    AssignedVariantAttribute = apps.get_model("attribute", "AssignedVariantAttribute")
+    AssignedVariantAttributeValue = apps.get_model(
+        "attribute", "AssignedVariantAttributeValue"
+    )
+    instances = []
+    for assignment in AssignedVariantAttribute.objects.all():
```
Comment (fowczarek) [inline saleor/attribute/migrations/0005_add_assignedattributevalues.py:27]: We should use iterator here. :policeman:

File: saleor/attribute/migrations/0005_add_assignedattributevalues.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    AssignedVariantAttributeValue.objects.bulk_create(instances)
+
+
+def create_through_page_relations(apps, schema_editor):
+    AssignedPageAttribute = apps.get_model("attribute", "AssignedPageAttribute")
+    AssignedPageAttributeValue = apps.get_model(
+        "attribute", "AssignedPageAttributeValue"
+    )
+    instances = []
+    for assignment in AssignedPageAttribute.objects.all():
```
Comment (fowczarek) [inline saleor/attribute/migrations/0005_add_assignedattributevalues.py:41]: We should use iterator here. :policeman:

File: saleor/graphql/page/mutations/attributes.py [hunk 4]
```diff
@@ -197,3 +200,87 @@ def perform_mutation(cls, _root, info, **data):
...
+            graphene.ID, required=True, description="ID of a page."
+        )
+        attribute_id = graphene.Argument(
+            graphene.ID, required=True, description="ID of an attribute."
+        )
+        moves = graphene.List(
+            ReorderInput,
+            required=True,
+            description="The list of reordering operations for given attribute values.",
+        )
```
Comment (fowczarek) [inline saleor/graphql/page/mutations/attributes.py:227]: Maybe we could move these attributes(`attribute_id` and `moves`) to `BaseReorderAttributeValuesMutation` ?

File: saleor/graphql/page/mutations/attributes.py [hunk 5]
```diff
@@ -197,3 +200,87 @@ def perform_mutation(cls, _root, info, **data):
...
+                    "page_id": ValidationError(
+                        (f"Couldn't resolve to a page: {page_id}"),
+                        code=PageErrorCode.NOT_FOUND.value,
+                    )
+                }
+            )
+        return product
+
+    @staticmethod
+    def get_attribute_assignment(page: page_models.Page, attribute_id: str):
```
Comment (fowczarek) [inline saleor/graphql/page/mutations/attributes.py:268]: Maybe we could move this method to `BaseReorderAttributeValuesMutation`?

File: saleor/graphql/page/mutations/attributes.py [hunk 6]
```diff
@@ -197,3 +200,87 @@ def perform_mutation(cls, _root, info, **data):
...
+            graphene.ID, required=True, description="ID of an attribute."
+        )
+        moves = graphene.List(
+            ReorderInput,
+            required=True,
+            description="The list of reordering operations for given attribute values.",
+        )
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
```
Comment (fowczarek) [inline saleor/graphql/page/mutations/attributes.py:230]: Maybe we can extract a lot of code from this method to `BaseReorderAttributeValuesMutation`?

File: saleor/graphql/page/mutations/attributes.py (hunk 4 above)
Comment (IKarbowiak) [reply saleor/graphql/page/mutations/attributes.py:227]: I left it as it's as it can't be moved to `BaseReorderAttributeValuesMutation`.

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**